xlators/mount/fuse/src/fuse-bridge.c

```diff
@@ -370,33 +381,108 @@ send_fuse_iov(xlator_t *this, fuse_in_header_t *finh, struct iovec *iov_out,
         fouh->len += iov_out[i].iov_len;
     fouh->unique = finh->unique;
 
-    res = sys_writev(priv->fd, iov_out, count);
-    gf_log("glusterfs-fuse", GF_LOG_TRACE, "writev() result %d/%d %s", res,
-           fouh->len, res == -1 ? strerror(errno) : "");
+    if (sync) {
+        res = sys_writev(priv->fd, iov_out, count);
+        gf_log("glusterfs-fuse", GF_LOG_TRACE, "writev() result %d/%d %s", res,
+               fouh->len, res == -1 ? strerror(errno) : "");
+    } else {
+        struct io_uring_sqe *sqe = NULL;
+
+        ctx = GF_MALLOC(sizeof(*ctx), gf_fuse_mt_uring_ctx);
+        if (!ctx) {
+            res = -ENOMEM;
+            gf_log("this->name", GF_LOG_ERROR, "Failed to allocate memory, %d",
+                   res);
+            goto err;
+        }
+
+        pthread_mutex_lock(&priv->sq_mutex);
+        {
+            sqe = io_uring_get_sqe(&priv->ring);
+            if (!sqe) {
+                pthread_mutex_unlock(&priv->sq_mutex);
+                gf_log("this->name", GF_LOG_ERROR,
+                       "Failed to get the submission queue for ring. Increase "
+                       "queue "
+                       "depth");
+                res = -1;
+                goto err;
+            }
 
+            io_uring_prep_writev(sqe, priv->fd, iov_out, count, /* offset */ 0);
+
+            sqe->flags |= IOSQE_ASYNC;
```

Comment:
I'm not sure if `IOSQE_ASYNC` makes sense for a write request. Do you have any reference ?

Reply:
I too have not found any example/usecase using IOSQE_ASYNC while doing write submission.
But using it by looking at this description:
https://unixism.net/loti/ref-iouring/io_uring_enter.html
IOSQE_ASYNC
Normal operation for io_uring is to try and issue an sqe as non-blocking first, and if that fails, execute it in an async manner. To support more efficient overlapped operation of requests that the application knows/assumes will always (or most of the time) block, the application can ask for an sqe to be issued async from the start. Available since 5.6.